### DISCUSSION TITLE
How to use WebGPU with eframe on the web?

### DISCUSSION THREAD

**Discussion by SGauvin** [emilk/egui#2870] [Q&A]
When using eframe, the chosen backed is WebGL.
Is there a way to ask it to use WebGPU instead? I'm trying to run compute shaders on Firefox Nightly inside an eframe application.

**Comment by stonerfish** [comment]
Remove the "glow" feature and add the "wgpu" feature to the eframe dependency in your Cargo.toml

**Reply by LubeckHuaman** [reply]
cannot run on browser after changing from glow to wgpu

**Reply by 9SMTM6** [reply]
> cannot run on browser after changing from glow to wgpu

Activate the webgl feature on wgpu.

Browsers don't support WebGPU on their stable releases yet - without flags  - , only exception is Chrome on Windows and perhaps MacOS, but not Linux.

With the webgl feature wgpu will fallback to translating your WebGPU calls to WebGL. Of course that means that performance will probably be worse than if you were using WebGL directly. But I personally don't want to touch any OpenGL derivative myself since my PTSD from Android.

**Comment by emilk** [comment]
Disable the `glow` feature and enable the `wgpu` feature. This is documented in https://docs.rs/eframe/latest/eframe/#feature-flags
Read the browser console logs to debug.
